geotf/src/geodetic_converter.cc

```diff
@@ -0,0 +1,436 @@
+  auto spatial_ref = std::make_shared<OGRSpatialReference>();
+  OGRErr err = spatial_ref->importFromEPSG(id);
+  if (err != OGRERR_NONE) {
+    std::cout << "ERROR" << err << std::endl;
+    return false;
+  }
+
+  mappings_.insert(std::make_pair(name, spatial_ref));
+
+  ROS_INFO_STREAM("[GeoTF] Added EPSG Code " << id <<
+                                             " as frame " << name);
+  return true;
+}
+
+// Add a coordinate frame by a Geo Coordinate System
+// Prominent example: WGS84
+bool GeodeticConverter::addFrameByGCSCode(const std::string& name,
+                                          const std::string& gcscode) {
+  if (mappings_.count(name)) {
+    return false;
+  }
+
+  // Create Spatial Reference from well known code
+  auto spatial_ref = std::make_shared<OGRSpatialReference>();
+  OGRErr err = spatial_ref->SetWellKnownGeogCS(gcscode.c_str());
+  if (err != OGRERR_NONE) {
+    std::cout << "ERROR" << err << std::endl;
+    return false;
+  }
+
+  mappings_.insert(std::make_pair(name, spatial_ref));
+  ROS_INFO_STREAM("[GeoTF] Added GCS Code " << gcscode <<
+                                            " as frame " << name);
+  return true;
+}
+
+// Add a frame by UTM Zone
+// zone is the UTM zone (e.g. switzerland is in Zone 32)
+// north is true for northern hemisphere zones.
+bool GeodeticConverter::addFramebyUTM(const std::string& name,
+                                      const uint zone,
+                                      const bool north) {
+
+  // Create Spatial Reference from well known code
+  auto spatial_ref = std::make_shared<OGRSpatialReference>();
+
+  spatial_ref->SetWellKnownGeogCS("WGS84");
+  spatial_ref->SetUTM(zone, north);
+
+  mappings_.insert(std::make_pair(name, spatial_ref));
+  ROS_INFO_STREAM("[GeoTF] Added UTM " << zone << "/"
+                                       << (north ? "N" : "S") <<
+                                       " as frame " << name);
+  return true;
+}
+
+// Writes a list of all frame definition to console
+void GeodeticConverter::writeDebugInfo() const {
+
+  for (auto key : mappings_) {
+    std::cout << key.first << std::endl;
+    char* pszWKT = NULL;
+    key.second->exportToWkt(&pszWKT);
+    printf("%s\n", pszWKT);
+    CPLFree(pszWKT);
+
+    std::cout << std::endl << std::endl;
+  }
+}
+
+// Creates a new ENU Frame with its origin at the given
+// Location (lon,lat, alt)
+// Where (lon,lat,alt) are defined w.r.t. WGS84
+bool GeodeticConverter::addFrameByENUOrigin(const std::string& name,
+                                            const double lon,
+                                            const double lat,
+                                            const double alt) {
+  // Create Spatial Reference from well known code
+  auto spatial_ref = std::make_shared<OGRSpatialReference>();
+
+  // ENU Frame based on GPS coordinates
+  spatial_ref->SetWellKnownGeogCS("WGS84");
+  spatial_ref->SetOrthographic(lat, lon, 0.0, 0.0);
```

Comment:
This is also defined as `latitude, longitude, ...` thus I would change the argument order of `addFrameByENUOrigin`